src/Features/Core/Portable/CodeFixes/Suppression/ConfigureSeverityLevelCodeAction.cs

```diff
@@ -0,0 +1,283 @@
+                {
+                    var id = DocumentId.CreateNewId(project.Id);
+                    var editorconfigDefaultFileContent =
+                        @"# ACTION REQUIRED: This file was automatically added to your project, but it
+# will not correctly take effect until additional steps are taken to enable it. See the
+# following page for additional information:
+#
+# https://github.com/dotnet/roslyn/blob/master/docs/analyzers/Enable%20Editorconfig%20Configuration.md";
+                    solution = solution.AddAdditionalDocument(id, ".editorconfig", editorconfigDefaultFileContent);
+                    editorconfig = solution.GetAdditionalDocument(id);
+                }
+            }
+
+            void FindRuleName()
+            {
+                if (diagnosticToEditorConfigDotNet.ContainsKey(diagnostic.Id))
+                {
+                    diagnosticToEditorConfigDotNet.TryGetValue(diagnostic.Id, out var value);
+                    var storageLocation = value.StorageLocations.OfType<EditorConfigStorageLocation<CodeStyleOption<bool>>>().FirstOrDefault();
+                    if (storageLocation != null)
+                    {
+                        name = storageLocation.KeyName;
+                    }
+
+                }
+                else if (languageSpecificOptions.ContainsKey(diagnostic.Id))
+                {
+                    languageSpecificOptions.TryGetValue(diagnostic.Id, out var value);
+                    var storageLocation = value.StorageLocations.OfType<EditorConfigStorageLocation<CodeStyleOption<bool>>>().FirstOrDefault();
+                    if (storageLocation != null)
+                    {
+                        name = storageLocation.KeyName;
+                    }
+                }
+                else if (expressionOptions != null && expressionOptions.ContainsKey(diagnostic.Id))
+                {
+                    expressionOptions.TryGetValue(diagnostic.Id, out var value);
+                    var storageLocation = value.StorageLocations.OfType<EditorConfigStorageLocation<CodeStyleOption<ExpressionBodyPreference>>>().FirstOrDefault();
+                    if (storageLocation != null)
+                    {
+                        name = storageLocation.KeyName;
+                    }
+                }
+                else if (diagnostic.Properties != null && diagnostic.Properties.ContainsKey("OptionName"))
+                {
+                    diagnostic.Properties.TryGetValue("OptionName", out name);
+                }
+            }
+
+            bool CheckIfRuleExistsInFile()
+            {
+                bool ruleFound = false;
+                foreach (var curLine in lines)
+                {
+                    var curLineText = curLine.ToString();
+
+                    var rulePattern = new Regex(@"([\w ]+)=([\w ]+):([\w ]+)");
+                    var headerPattern = new Regex(@"\[\*([^#\[]*)\]");
+                    if (rulePattern.IsMatch(curLineText))
```

Comment:
You should probably add a few unit tests where the header or rule entry has typo.